### ISSUE TITLE
As a developer I want to be able to PATCH or PUT rollen in the Zaken API

### ISSUE THREAD

**Issue by damm89** [VNG-Realisatie/gemma-zaken#1835]
**Summary**:
To reduce the number of requests it would be great if we can PATCH or PUT `rollen` (and other resources).

**Context**:
In some cases a `rol` is created at a time where more information on the `rol` is unknown. In our particular implementation we're using a BPMN modeler (Camunda) so that (non-)developers can create resources such as `rollen` at particular points in time in automated processes. The BPMN modeler is naive in the sense that it doesn't hold all information of every user and polluting it with that data is undesirable. Currently a `rol` gets created in the BPMN modeler but with sparse data (only the `username` is added to the `betrokkeneIdentificatie`).

**Current implementation**:
A `rol` resource can be created or destroyed.

**Current inefficiencies**:
A notification from the NRC alerts our Zaakafhandelcomponent that a `rol` is created. This sets in motion the processes to complete the information in the betrokkeneIdentificatie by:

1. getting the `rol`
2. temporarily storing the `rol` information 
3. marrying other data to the `rol` if other data exists
4. destroying the `rol` (alerts the NRC)
5. creating a new `rol` (alerts the NRC) with updated data

**Benefits**:
The number of requests related to "editing" these resources would be halved.

**Comment by edgarvonk** [comment]
We are working on the development of [ZAC](https://github.com/infonl/dimpact-zaakafhandelcomponent) and we would also very much like to be able to change (zaak)rollen with just one ZGW ZRC API request (PUT or possibly PATCH) instead of the two we now need to do (DELETE, followed by a CREATE).

In ZAC users can (re-)assign ('verdelen') batches of up to 100 zaken in one go. This action results in _a lot_ of ZGW API traffic to the zaakregister (OpenZaak in our case) and this is for a large part because we need to first delete and then create these zaakrollen for every zaak. And also, in the ZGW architecture this currently means that we receive and need to handle two notifications (from OpenNotificaties in our case): one for the zaakrol delete and one for the zaakrol create while we really are only interested in a single notificatie: zaakrole update.

In other words: we find that the current ZGW ZRC API is too low-level for this (and possibly other) use case and we (and I think other zaakafhandelcomponents) would greatly benefit from an addition to this API to be able to update zaakrollen in one request.

The following JIRA issue contains some more information specific for ZAC: https://dimpact.atlassian.net/browse/PZ-4371

**Comment by Hugo-ter-Doest** [comment]
Thanks Edgar!
@joeridiederen this aligns very well with the G4 initiative to create handelinggedreven API's in addition to the CRUD API's.

**Comment by edgarvonk** [comment]
@Hugo-ter-Doest there may well be more places than just this where ZAC (and other components) would greatly benefit from higher-level handelinggedreven API's as compared to the current CRUD ones. Might be worthwhile to do an investigation at some point?

**Comment by HenriKorver** [comment]
@edgarvonk Do you really have an urgent (performance) problem right now? If so, does it solve the problem for you if we only add the PUT-operation to the `/rollen` resource and not the PATCH?

**Comment by edgarvonk** [comment]
> @edgarvonk Do you really have an urgent (performance) problem right now? If so, does it solve the problem for you if we only add the PUT-operation to the `/rollen` resource and not the PATCH?

Only the PUT would be fine! Actually I would prefer that even just to keep things simple. No, we don't have an urgent performance problem. Our main driver for this change is more to reduce complexity and improve reliability of our solution. But performance gains are always welcome for sure. :-)

**Comment by HenriKorver** [comment]
See PR #2489. You can view the addition of the PUT-operation to the `/rollen` endpoint  [here](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/VNG-Realisatie/gemma-zaken/zrc-rollen-update/api-specificatie/zrc/1.5.x/1.5.2/openapi.yaml&nocors#tag/rollen/operation/rol_update) with the Redoc Viewer.

@edgarvonk Does this satisfy your needs?

**Comment by HenriKorver** [comment]
Note that in this definition you can not change the attributes: `zaak`, `betrokkene` and `brokkeneType`.

![afbeelding](https://github.com/user-attachments/assets/e6b0cce2-7b8e-4bfe-abcc-e3fc1692271e)

**Comment by marnixdessing** [comment]
Is there any reason why the `betrokkene` field is not updatable? As it is a non-required attribute is would be nice to set it at a later point in time. At Gemeente Nijmegen we have a use case for doing so. We use this field to store a URL reference (to an external system), however the object to be referenced is created after the role is created. Hence, we need to update the `betrokkene` field.

**Comment by HenriKorver** [comment]
The reason was that I wanted  to be consistent with the existing PUT-operation at the `/zaakobjecten` endpoint:

![afbeelding](https://github.com/user-attachments/assets/cb19c080-ce8f-4dc2-a4df-c6cbb6da5c9a)

But if you want to update the `betrokkene` field I can change the specification.

**Comment by HenriKorver** [comment]
@marnixdessing In commit https://github.com/VNG-Realisatie/gemma-zaken/pull/2489/commits/7492f1a382adde4337e54833be742f3a173f458a of the PR I have relaxed on the validation such that you can update the `betrokkene` field. Is this to your liking?

**Comment by marnixdessing** [comment]
@HenriKorver thanks! This will work for us :)

**Comment by johannesbattjes** [comment]
@HenriKorver hier staat dat Put is toegevoegd aan Rol in ZGW 1.6. 
Maar ik zie hem niet in de OAS: https://vng-realisatie.github.io/gemma-zaken/standaard/zaken/redoc-1.6.0#tag/rollen
Kun jij uitleggen wat er wel/niet in ZRC 1.6 zou moeten zitten?
Motivatie: dit probleem is voor ons ook belangrijk (aangemeld als onderdeel van #2420 )

**Comment by HenriKorver** [comment]
@johannesbattjes Blijkbaar zijn we dit issue vergeten mee te nemen in de ZGW 1.6 release. Het is niet eerder opgemerkt tijdens de consultatie. Ik zal het issue heropenen.

**Comment by johannesbattjes** [comment]
@HenriKorver herinnering. Hij zit er nog steeds niet in lijkt het

<img width="838" height="562" alt="Image" src="https://github.com/user-attachments/assets/cdfece03-5991-435f-8d0a-2feab817b3e7" />

**Comment by HenriKorver** [comment]
@johannesbattjes Bedankt voor de observatie, dit issue is inderdaad tussen wal en schip gevallen. Ik heb het heropend en we zullen het meenemen in 1.8.
